### ISSUE TITLE
Boolean kwargs to `@dataclass` on `NamedTuple` reports confusing errors

### ISSUE THREAD

**Issue by bzoracler** [python/mypy#15338]
**Bug Report & To Reproduce**

I realise it's probably meaningless to have `@dataclasses` on a `NamedTuple`, but this behaviour was rather unexpected:

```python
from typing import NamedTuple
from dataclasses import dataclass

@dataclass(
    init=True,  # error: "init" argument must be a True or False literal  [literal-required]
    repr=False,  # No error here
    eq=False,  # error: "eq" argument must be a True or False literal  [literal-required]
    kw_only=True,  # error: "kw_only" argument must be a True or False literal  [literal-required]
    slots=True,  # error: "slots" argument must be a True or False literal  [literal-required]
)
class Data(NamedTuple): ...
```

**Expected Behavior**

Not this - or a warning when decorating `NamedTuple` with `@dataclass`. Also, I'm not sure why `repr=False` doesn't cause an error.

**Your Environment**

gist: https://mypy-play.net/?mypy=latest&python=3.11&gist=109b7d68558966d96c4063b2dcc49e6a

- Mypy version used: 1.3.0
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.10 & 3.11

**Comment by sobolevn** [comment]
For some reason not all `True` and `False` name expressions have bound `fullname`. This why this error happens. Probably, we have some problems during the semantic analysis.

**Comment by erictraut** [comment]
This bug appears to have been fixed in mypy 1.5.

**Comment by sobolevn** [comment]
Confirmed.
